core/src/main/java/org/eclipse/krazo/lifecycle/RequestLifecycle.java

```diff
@@ -51,12 +54,12 @@ public void beforeAll(ContainerRequestContext context) {
 
     }
 
-    public Object aroundController(Callable<Object> controllerMethod) throws Exception {
+    public Object aroundController(final Method invokedMethod, Callable<Object> controllerMethod) throws Exception {
 
         eventDispatcher.fireBeforeControllerEvent();
         try {
-
             Object result = controllerMethod.call();
+            this.invokedMethod = invokedMethod;
```

Comment:
I'm wondering if we should move this line up, so that we have access to `invokedMethod` before the controller is invoked and maybe even before the `BeforeController` events are fired. In this case we should prefer to call the field `controllerMethod` and not `invokedControllerMethod` because it is available before the controller has been invoked.